src/ast-to-react.js

```diff
@@ -136,8 +132,8 @@ exports.hastChildrenToReact = childrenToReact
  * @property {TableRowComponent|ReactMarkdownNames} tr
  * @property {UnorderedListComponent|ReactMarkdownNames} ul
  *
- * @typedef {Record<Exclude<ReactMarkdownNames, keyof SpecialComponents>, NormalComponent|ReactMarkdownNames>} NormalComponents
- * @typedef {Partial<NormalComponents & SpecialComponents>} Components
+ * @typedef {{ [P in keyof IntrinsicElements]: P | ((props: IntrinsicElements[P] & ReactMarkdownProps) => ReactNode) }} NormalComponents
```

Comment:
```suggestion
 * @typedef {{[TagName in keyof IntrinsicElements]: TagName | ((props: IntrinsicElements[TagName] & ReactMarkdownProps) => ReactNode)}} NormalComponents
```

Matches style and uses a more clear variable name. Open to other names.
Otherwise, I think this is good